src/parser/core/modules/Cooldowns.js

```diff
@@ -188,7 +188,12 @@ export default class Cooldowns extends Module {
 		// If there's a current object, move it into the history
 		// TODO: handle errors on CD overlap
 		if (cd.current) {
-			cd.history.push(cd.current)
+			if (cd.current.timestamp < this.parser.fight.start_time && cd.length * 1000 > this.parser.currentTimestamp) {
```

Comment:
```suggestion
			if (cd.current.timestamp < this.parser.fight.start_time && action.cooldown * 1000 > this.parser.currentTimestamp - cd.current.timestamp) {
```
This is bugged. Suggestion should fix it.

Reply:
I went with a slightly different approach, to be more explicit that I'm checking against fight duration for the first post-pull use, but should be the same result.